homeassistant/components/balboa/climate.py

```diff
@@ -0,0 +1,188 @@
+"""Support for Balboa Spa Wifi adaptor."""
+from __future__ import annotations
+
+import math
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    FAN_OFF,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    SUPPORT_FAN_MODE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import (
+    ATTR_TEMPERATURE,
+    PRECISION_HALVES,
+    PRECISION_WHOLE,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+
+from .balboa_entity import BalboaEntity
+from .const import CLIMATE, CLIMATE_SUPPORTED_FANSTATES, CLIMATE_SUPPORTED_MODES, DOMAIN
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the spa climate device."""
+    async_add_entities(
+        [
+            BalboaSpaClimate(
+                hass,
+                entry,
+                hass.data[DOMAIN][entry.entry_id],
+                CLIMATE,
+            )
+        ],
+        True,
+    )
+
+
+class BalboaSpaClimate(BalboaEntity, ClimateEntity):
+    """Representation of a Balboa Spa Climate device."""
+
+    _attr_icon = "mdi:hot-tub"
+    _attr_fan_modes = CLIMATE_SUPPORTED_FANSTATES
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        features = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+
+        if self._client.have_blower():
+            features |= SUPPORT_FAN_MODE
+
+        return features
```

Comment:
Generally we prefer to set this as `self._attr_supported_features` in `__init__` so the code doesn't have to be run every time the entity state is written.

The same is true for `hvac_modes` , `preset_modes`, and any other property not expected to change during the lifecycle of the entity.